src/tests/multihost/alltests/test_inmemory_cache.py

```diff
@@ -0,0 +1,355 @@
+        find_logs(multihost, log_name, "/var/lib/sss/mc/passwd")
+        find_logs(multihost, log_ssd, "Got request for ")
+        find_logs(multihost, log_ssd, "uid=foo1")
+        clear_only_domain_log(multihost)
+        client.run_command("strace -o /tmp/stracegroup1 getent group ldapusers")
+        log_name = "/tmp/stracegroup1"
+        find_logs(multihost, log_name, "/var/lib/sss/pipes/nss")
+        find_logs(multihost, log_name, "/var/lib/sss/mc/group")
+        find_logs(multihost, log_ssd, "Got request for ")
+        find_logs(multihost, log_ssd, "cn=ldapusers")
+
+    @staticmethod
+    def test_maxage(multihost, common_sssd_setup):
+        """
+        :title: Data is directly pulled from inmemory cache
+            without opening the pipe to the NSS responder
+        :id: 31217168-96e9-11ed-b69f-845cf3eff344
+        :steps:
+            1. Clear sssd domain logs
+            2. Make sure /var/lib/sss/pipes/nss was not called while cache is there
+            3. sssd should not send request to master server while cache is there,
+                by checking that new request was not sent to master server.
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+        """
+        client = multihost.client[0]
+        file_ssd = f'/var/log/sssd/sssd_{ds_instance_name}.log'
+        client.run_command("strace -o /tmp/straceuser1 getent passwd foo1")
+        client.run_command("strace -o /tmp/stracegroup1 getent group ldapusers")
+        clear_only_domain_log(multihost)
+        client.run_command("strace -o /tmp/straceuser2 getent passwd foo1")
+        with pytest.raises(AssertionError):
+            find_logs(multihost, "/tmp/straceuser2", "/var/lib/sss/pipes/nss")
+        with pytest.raises(AssertionError):
+            find_logs(multihost, file_ssd, "Got request for ")
+        with pytest.raises(AssertionError):
+            find_logs(multihost, file_ssd, "uid=foo1")
+        find_logs(multihost, "/tmp/straceuser2", "/var/lib/sss/mc/passwd")
+        clear_only_domain_log(multihost)
+        client.run_command("strace -o /tmp/stracegroup2 getent group ldapusers")
+        with pytest.raises(AssertionError):
+            find_logs(multihost, "/tmp/stracegroup2", "/var/lib/sss/pipes/nss")
+        with pytest.raises(AssertionError):
+            find_logs(multihost, file_ssd, "Got request for ")
+        with pytest.raises(AssertionError):
+            find_logs(multihost, file_ssd, "cn=ldapusers")
+        find_logs(multihost, "/tmp/stracegroup2", "/var/lib/sss/mc/group")
+
+    @staticmethod
+    def test_bz954323(multihost, common_sssd_setup):
+        """
+        :title: sss cache invalidates the InMemory cache
+        :id: 92ee87f8-96eb-11ed-a172-845cf3eff344
+        :steps:
+            1. Make sure sss cache invalidates the InMemory cache.
+        :expectedresults:
+            1. Force invalidation of cached data and thus
```

Comment:
Steps and expectedresults do not match. Could step1 from expectedresults be moved to steps.